File: src/types/local_skill.ts
export interface LocalPluginStatus {
  marketplace_path: string;
  marketplace_repo: string;
  tracked_path: string;
  installed_path: string;
  tracked_exists: boolean;
  installed_exists: boolean;
  up_to_date: boolean;
  tracked_file_count: number;
  installed_file_count: number;
  registered_agent_count: number;
  generated_agents: string[];
  missing_files: string[];
  changed_files: string[];
}

export interface LocalPluginSyncResult {
  status: LocalPluginStatus;
  copied_files: string[];
  removed_files: string[];
  preserved_files: string[];
}

export interface CcSwitchDeeplink {
  url: string;
  resource_type: string;
  description: string;
}

export interface CcSwitchDeeplinkResult {
  provider_link: CcSwitchDeeplink | null;
  skill_link: CcSwitchDeeplink | null;
}

File: src-tauri/src/domain/local_skill.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LocalPluginStatus {
    pub marketplace_path: String,
    pub marketplace_repo: String,
    pub tracked_path: String,
    pub installed_path: String,
    pub tracked_exists: bool,
    pub installed_exists: bool,
    pub up_to_date: bool,
    pub tracked_file_count: usize,
    pub installed_file_count: usize,
    pub missing_files: Vec<String>,
    pub changed_files: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LocalPluginSyncResult {
    pub status: LocalPluginStatus,
    pub copied_files: Vec<String>,
    pub removed_files: Vec<String>,
    pub preserved_files: Vec<String>,
}
